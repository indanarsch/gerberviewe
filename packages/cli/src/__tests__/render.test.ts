// Tests for @tracespace/cli
import path from 'node:path'
import fs from 'node:fs/promises'

import {temporaryDirectory} from 'tempy'
import {describe, it, beforeEach, afterEach, expect} from 'vitest'

import {run} from './run'

const FIXTURES = [
  'pads/circle.gbr',
  'pads/obround.gbr',
  'pads/polygon.gbr',
  'pads/rectangle.gbr',
]

const getGerberFixturePath = (name: string) => {
  return require.resolve(`@tracespace/fixtures/gerbers/${name}`)
}

const readFile = async (file: string) => {
  return fs.readFile(file, 'utf8')
}

const readJson = async (file: string) => {
  return fs
    .readFile(file, 'utf8')
    .then(contents => JSON.parse(contents) as Record<string, unknown>)
}

describe('@tracespace/cli render', () => {
  let outDir: string

  beforeEach(() => {
    outDir = temporaryDirectory({prefix: 'tracespace'})
  })

  afterEach(async () => {
    return fs.rm(outDir, {force: true, recursive: true})
  })

  FIXTURES.forEach(fixtureName => {
    const source = getGerberFixturePath(fixtureName)

    it(`should render ${fixtureName}`, async () => {
      const outName = path.join(outDir, path.basename(source))
      const parseFile = `${outName}.parse.json`
      const plotFile = `${outName}.plot.json`
      const renderFile = `${outName}.svg`

      await run('render', source, '--out', outDir, '--debug')

      return Promise.all([
        readJson(parseFile),
        readJson(plotFile),
        readFile(renderFile),
      ]).then(([parseResults, plotResults, renderResults]) => {
        expect(parseResults).toMatchSnapshot('parse')
        expect(plotResults).toMatchSnapshot('plot')
        expect(renderResults).toMatchSnapshot('render')
      })
    })
  })
})
