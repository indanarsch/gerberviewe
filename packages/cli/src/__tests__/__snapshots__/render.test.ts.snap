// Vitest Snapshot v1

exports[`@tracespace/cli render > should render pads/circle.gbr > parse 1`] = `
{
  "children": [
    {
      "format": [
        3,
        4,
      ],
      "mode": "absolute",
      "position": {
        "end": {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "coordinateFormat",
      "zeroSuppression": "leading",
    },
    {
      "position": {
        "end": {
          "column": 6,
          "line": 2,
          "offset": 19,
        },
        "start": {
          "column": 2,
          "line": 2,
          "offset": 15,
        },
      },
      "type": "units",
      "units": "in",
    },
    {
      "code": "10",
      "hole": null,
      "position": {
        "end": {
          "column": 12,
          "line": 3,
          "offset": 33,
        },
        "start": {
          "column": 2,
          "line": 3,
          "offset": 23,
        },
      },
      "shape": {
        "diameter": 0.5,
        "type": "circle",
      },
      "type": "toolDefinition",
    },
    {
      "code": "10",
      "position": {
        "end": {
          "column": 4,
          "line": 4,
          "offset": 39,
        },
        "start": {
          "column": 1,
          "line": 4,
          "offset": 36,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "0",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 8,
          "line": 5,
          "offset": 48,
        },
        "start": {
          "column": 1,
          "line": 5,
          "offset": 41,
        },
      },
      "type": "graphic",
    },
    {
      "position": {
        "end": {
          "column": 4,
          "line": 6,
          "offset": 53,
        },
        "start": {
          "column": 1,
          "line": 6,
          "offset": 50,
        },
      },
      "type": "done",
    },
  ],
  "filetype": "gerber",
  "type": "root",
}
`;

exports[`@tracespace/cli render > should render pads/circle.gbr > plot 1`] = `
{
  "children": [
    {
      "children": [
        {
          "shape": {
            "cx": 0,
            "cy": 0,
            "r": 0.25,
            "type": "circle",
          },
          "type": "imageShape",
        },
      ],
      "size": [
        -0.25,
        -0.25,
        0.25,
        0.25,
      ],
      "type": "imageLayer",
    },
  ],
  "type": "image",
  "units": "in",
}
`;

exports[`@tracespace/cli render > should render pads/circle.gbr > render 1`] = `"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"0\\" fill-rule=\\"evenodd\\" fill=\\"currentColor\\" stroke=\\"currentColor\\" width=\\"0.5in\\" height=\\"0.5in\\" viewBox=\\"-0.25 -0.25 0.5 0.5\\"><g transform=\\"translate(0, 0) scale(1,-1)\\"><circle cx=\\"0\\" cy=\\"0\\" r=\\"0.25\\"></circle></g></svg>"`;

exports[`@tracespace/cli render > should render pads/obround.gbr > parse 1`] = `
{
  "children": [
    {
      "format": [
        3,
        4,
      ],
      "mode": "absolute",
      "position": {
        "end": {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "coordinateFormat",
      "zeroSuppression": "leading",
    },
    {
      "position": {
        "end": {
          "column": 6,
          "line": 2,
          "offset": 19,
        },
        "start": {
          "column": 2,
          "line": 2,
          "offset": 15,
        },
      },
      "type": "units",
      "units": "in",
    },
    {
      "code": "10",
      "hole": null,
      "position": {
        "end": {
          "column": 16,
          "line": 3,
          "offset": 37,
        },
        "start": {
          "column": 2,
          "line": 3,
          "offset": 23,
        },
      },
      "shape": {
        "type": "obround",
        "xSize": 1,
        "ySize": 0.5,
      },
      "type": "toolDefinition",
    },
    {
      "code": "10",
      "position": {
        "end": {
          "column": 4,
          "line": 4,
          "offset": 43,
        },
        "start": {
          "column": 1,
          "line": 4,
          "offset": 40,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "0",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 8,
          "line": 5,
          "offset": 52,
        },
        "start": {
          "column": 1,
          "line": 5,
          "offset": 45,
        },
      },
      "type": "graphic",
    },
    {
      "position": {
        "end": {
          "column": 4,
          "line": 6,
          "offset": 57,
        },
        "start": {
          "column": 1,
          "line": 6,
          "offset": 54,
        },
      },
      "type": "done",
    },
  ],
  "filetype": "gerber",
  "type": "root",
}
`;

exports[`@tracespace/cli render > should render pads/obround.gbr > plot 1`] = `
{
  "children": [
    {
      "children": [
        {
          "shape": {
            "r": 0.25,
            "type": "rectangle",
            "x": -0.5,
            "xSize": 1,
            "y": -0.25,
            "ySize": 0.5,
          },
          "type": "imageShape",
        },
      ],
      "size": [
        -0.5,
        -0.25,
        0.5,
        0.25,
      ],
      "type": "imageLayer",
    },
  ],
  "type": "image",
  "units": "in",
}
`;

exports[`@tracespace/cli render > should render pads/obround.gbr > render 1`] = `"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"0\\" fill-rule=\\"evenodd\\" fill=\\"currentColor\\" stroke=\\"currentColor\\" width=\\"1in\\" height=\\"0.5in\\" viewBox=\\"-0.5 -0.25 1 0.5\\"><g transform=\\"translate(0, 0) scale(1,-1)\\"><rect x=\\"-0.5\\" y=\\"-0.25\\" width=\\"1\\" height=\\"0.5\\" rx=\\"0.25\\" ry=\\"0.25\\"></rect></g></svg>"`;

exports[`@tracespace/cli render > should render pads/polygon.gbr > parse 1`] = `
{
  "children": [
    {
      "format": [
        3,
        4,
      ],
      "mode": "absolute",
      "position": {
        "end": {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "coordinateFormat",
      "zeroSuppression": "leading",
    },
    {
      "position": {
        "end": {
          "column": 6,
          "line": 2,
          "offset": 19,
        },
        "start": {
          "column": 2,
          "line": 2,
          "offset": 15,
        },
      },
      "type": "units",
      "units": "in",
    },
    {
      "code": "10",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 3,
          "offset": 35,
        },
        "start": {
          "column": 2,
          "line": 3,
          "offset": 23,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 3,
      },
      "type": "toolDefinition",
    },
    {
      "code": "11",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 4,
          "offset": 51,
        },
        "start": {
          "column": 2,
          "line": 4,
          "offset": 39,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 4,
      },
      "type": "toolDefinition",
    },
    {
      "code": "12",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 5,
          "offset": 67,
        },
        "start": {
          "column": 2,
          "line": 5,
          "offset": 55,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 5,
      },
      "type": "toolDefinition",
    },
    {
      "code": "13",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 6,
          "offset": 83,
        },
        "start": {
          "column": 2,
          "line": 6,
          "offset": 71,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 6,
      },
      "type": "toolDefinition",
    },
    {
      "code": "14",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 7,
          "offset": 99,
        },
        "start": {
          "column": 2,
          "line": 7,
          "offset": 87,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 7,
      },
      "type": "toolDefinition",
    },
    {
      "code": "15",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 8,
          "offset": 115,
        },
        "start": {
          "column": 2,
          "line": 8,
          "offset": 103,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 8,
      },
      "type": "toolDefinition",
    },
    {
      "code": "16",
      "hole": null,
      "position": {
        "end": {
          "column": 14,
          "line": 9,
          "offset": 131,
        },
        "start": {
          "column": 2,
          "line": 9,
          "offset": 119,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 9,
      },
      "type": "toolDefinition",
    },
    {
      "code": "17",
      "hole": null,
      "position": {
        "end": {
          "column": 15,
          "line": 10,
          "offset": 148,
        },
        "start": {
          "column": 2,
          "line": 10,
          "offset": 135,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 10,
      },
      "type": "toolDefinition",
    },
    {
      "code": "18",
      "hole": null,
      "position": {
        "end": {
          "column": 15,
          "line": 11,
          "offset": 165,
        },
        "start": {
          "column": 2,
          "line": 11,
          "offset": 152,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 11,
      },
      "type": "toolDefinition",
    },
    {
      "code": "19",
      "hole": null,
      "position": {
        "end": {
          "column": 15,
          "line": 12,
          "offset": 182,
        },
        "start": {
          "column": 2,
          "line": 12,
          "offset": 169,
        },
      },
      "shape": {
        "diameter": 0.5,
        "rotation": null,
        "type": "polygon",
        "vertices": 12,
      },
      "type": "toolDefinition",
    },
    {
      "code": "10",
      "position": {
        "end": {
          "column": 4,
          "line": 13,
          "offset": 188,
        },
        "start": {
          "column": 1,
          "line": 13,
          "offset": 185,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "0",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 8,
          "line": 14,
          "offset": 197,
        },
        "start": {
          "column": 1,
          "line": 14,
          "offset": 190,
        },
      },
      "type": "graphic",
    },
    {
      "code": "11",
      "position": {
        "end": {
          "column": 4,
          "line": 15,
          "offset": 202,
        },
        "start": {
          "column": 1,
          "line": 15,
          "offset": 199,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "6000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 11,
          "line": 16,
          "offset": 214,
        },
        "start": {
          "column": 1,
          "line": 16,
          "offset": 204,
        },
      },
      "type": "graphic",
    },
    {
      "code": "12",
      "position": {
        "end": {
          "column": 4,
          "line": 17,
          "offset": 219,
        },
        "start": {
          "column": 1,
          "line": 17,
          "offset": 216,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "12000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 18,
          "offset": 232,
        },
        "start": {
          "column": 1,
          "line": 18,
          "offset": 221,
        },
      },
      "type": "graphic",
    },
    {
      "code": "13",
      "position": {
        "end": {
          "column": 4,
          "line": 19,
          "offset": 237,
        },
        "start": {
          "column": 1,
          "line": 19,
          "offset": 234,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "18000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 20,
          "offset": 250,
        },
        "start": {
          "column": 1,
          "line": 20,
          "offset": 239,
        },
      },
      "type": "graphic",
    },
    {
      "code": "14",
      "position": {
        "end": {
          "column": 4,
          "line": 21,
          "offset": 255,
        },
        "start": {
          "column": 1,
          "line": 21,
          "offset": 252,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "24000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 22,
          "offset": 268,
        },
        "start": {
          "column": 1,
          "line": 22,
          "offset": 257,
        },
      },
      "type": "graphic",
    },
    {
      "code": "15",
      "position": {
        "end": {
          "column": 4,
          "line": 23,
          "offset": 273,
        },
        "start": {
          "column": 1,
          "line": 23,
          "offset": 270,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "30000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 24,
          "offset": 286,
        },
        "start": {
          "column": 1,
          "line": 24,
          "offset": 275,
        },
      },
      "type": "graphic",
    },
    {
      "code": "16",
      "position": {
        "end": {
          "column": 4,
          "line": 25,
          "offset": 291,
        },
        "start": {
          "column": 1,
          "line": 25,
          "offset": 288,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "36000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 26,
          "offset": 304,
        },
        "start": {
          "column": 1,
          "line": 26,
          "offset": 293,
        },
      },
      "type": "graphic",
    },
    {
      "code": "17",
      "position": {
        "end": {
          "column": 4,
          "line": 27,
          "offset": 309,
        },
        "start": {
          "column": 1,
          "line": 27,
          "offset": 306,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "42000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 28,
          "offset": 322,
        },
        "start": {
          "column": 1,
          "line": 28,
          "offset": 311,
        },
      },
      "type": "graphic",
    },
    {
      "code": "18",
      "position": {
        "end": {
          "column": 4,
          "line": 29,
          "offset": 327,
        },
        "start": {
          "column": 1,
          "line": 29,
          "offset": 324,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "48000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 30,
          "offset": 340,
        },
        "start": {
          "column": 1,
          "line": 30,
          "offset": 329,
        },
      },
      "type": "graphic",
    },
    {
      "code": "19",
      "position": {
        "end": {
          "column": 4,
          "line": 31,
          "offset": 345,
        },
        "start": {
          "column": 1,
          "line": 31,
          "offset": 342,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "54000",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 12,
          "line": 32,
          "offset": 358,
        },
        "start": {
          "column": 1,
          "line": 32,
          "offset": 347,
        },
      },
      "type": "graphic",
    },
    {
      "position": {
        "end": {
          "column": 4,
          "line": 33,
          "offset": 363,
        },
        "start": {
          "column": 1,
          "line": 33,
          "offset": 360,
        },
      },
      "type": "done",
    },
  ],
  "filetype": "gerber",
  "type": "root",
}
`;

exports[`@tracespace/cli render > should render pads/polygon.gbr > plot 1`] = `
{
  "children": [
    {
      "children": [
        {
          "shape": {
            "points": [
              [
                0.25,
                0,
              ],
              [
                -0.125,
                0.2165063509,
              ],
              [
                -0.125,
                -0.2165063509,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                0.6,
              ],
              [
                0,
                0.85,
              ],
              [
                -0.25,
                0.6,
              ],
              [
                0,
                0.35,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                1.2,
              ],
              [
                0.0772542486,
                1.4377641291,
              ],
              [
                -0.2022542486,
                1.3469463131,
              ],
              [
                -0.2022542486,
                1.0530536869,
              ],
              [
                0.0772542486,
                0.9622358709,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                1.8,
              ],
              [
                0.125,
                2.0165063509,
              ],
              [
                -0.125,
                2.0165063509,
              ],
              [
                -0.25,
                1.8,
              ],
              [
                -0.125,
                1.5834936491,
              ],
              [
                0.125,
                1.5834936491,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                2.4,
              ],
              [
                0.1558724505,
                2.5954578706,
              ],
              [
                -0.0556302335,
                2.643731978,
              ],
              [
                -0.225242217,
                2.5084709348,
              ],
              [
                -0.225242217,
                2.2915290652,
              ],
              [
                -0.0556302335,
                2.156268022,
              ],
              [
                0.1558724505,
                2.2045421294,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                3,
              ],
              [
                0.1767766953,
                3.1767766953,
              ],
              [
                0,
                3.25,
              ],
              [
                -0.1767766953,
                3.1767766953,
              ],
              [
                -0.25,
                3,
              ],
              [
                -0.1767766953,
                2.8232233047,
              ],
              [
                0,
                2.75,
              ],
              [
                0.1767766953,
                2.8232233047,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                3.6,
              ],
              [
                0.1915111108,
                3.7606969024,
              ],
              [
                0.0434120444,
                3.8462019383,
              ],
              [
                -0.125,
                3.8165063509,
              ],
              [
                -0.2349231552,
                3.6855050358,
              ],
              [
                -0.2349231552,
                3.5144949642,
              ],
              [
                -0.125,
                3.3834936491,
              ],
              [
                0.0434120444,
                3.3537980617,
              ],
              [
                0.1915111108,
                3.4393030976,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                4.2,
              ],
              [
                0.2022542486,
                4.3469463131,
              ],
              [
                0.0772542486,
                4.4377641291,
              ],
              [
                -0.0772542486,
                4.4377641291,
              ],
              [
                -0.2022542486,
                4.3469463131,
              ],
              [
                -0.25,
                4.2,
              ],
              [
                -0.2022542486,
                4.0530536869,
              ],
              [
                -0.0772542486,
                3.9622358709,
              ],
              [
                0.0772542486,
                3.9622358709,
              ],
              [
                0.2022542486,
                4.0530536869,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                4.8,
              ],
              [
                0.2103133832,
                4.9351602044,
              ],
              [
                0.1038537533,
                5.0274079988,
              ],
              [
                -0.0355787096,
                5.0474553605,
              ],
              [
                -0.1637151835,
                4.9889373936,
              ],
              [
                -0.2398732434,
                4.8704331392,
              ],
              [
                -0.2398732434,
                4.7295668608,
              ],
              [
                -0.1637151835,
                4.6110626064,
              ],
              [
                -0.0355787096,
                4.5525446395,
              ],
              [
                0.1038537533,
                4.5725920012,
              ],
              [
                0.2103133832,
                4.6648397956,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
        {
          "shape": {
            "points": [
              [
                0.25,
                5.4,
              ],
              [
                0.2165063509,
                5.525,
              ],
              [
                0.125,
                5.6165063509,
              ],
              [
                0,
                5.65,
              ],
              [
                -0.125,
                5.6165063509,
              ],
              [
                -0.2165063509,
                5.525,
              ],
              [
                -0.25,
                5.4,
              ],
              [
                -0.2165063509,
                5.275,
              ],
              [
                -0.125,
                5.1834936491,
              ],
              [
                0,
                5.15,
              ],
              [
                0.125,
                5.1834936491,
              ],
              [
                0.2165063509,
                5.275,
              ],
            ],
            "type": "polygon",
          },
          "type": "imageShape",
        },
      ],
      "size": [
        -0.25,
        -0.2165063509,
        0.25,
        5.65,
      ],
      "type": "imageLayer",
    },
  ],
  "type": "image",
  "units": "in",
}
`;

exports[`@tracespace/cli render > should render pads/polygon.gbr > render 1`] = `"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"0\\" fill-rule=\\"evenodd\\" fill=\\"currentColor\\" stroke=\\"currentColor\\" width=\\"0.5in\\" height=\\"5.8665063509in\\" viewBox=\\"-0.25 -0.2165063509 0.5 5.8665063509\\"><g transform=\\"translate(0, 5.4334936491) scale(1,-1)\\"><polygon points=\\"0.25,0 -0.125,0.2165063509 -0.125,-0.2165063509\\"></polygon><polygon points=\\"0.25,0.6 0,0.85 -0.25,0.6 0,0.35\\"></polygon><polygon points=\\"0.25,1.2 0.0772542486,1.4377641291 -0.2022542486,1.3469463131 -0.2022542486,1.0530536869 0.0772542486,0.9622358709\\"></polygon><polygon points=\\"0.25,1.8 0.125,2.0165063509 -0.125,2.0165063509 -0.25,1.8 -0.125,1.5834936491 0.125,1.5834936491\\"></polygon><polygon points=\\"0.25,2.4 0.1558724505,2.5954578706 -0.0556302335,2.643731978 -0.225242217,2.5084709348 -0.225242217,2.2915290652 -0.0556302335,2.156268022 0.1558724505,2.2045421294\\"></polygon><polygon points=\\"0.25,3 0.1767766953,3.1767766953 0,3.25 -0.1767766953,3.1767766953 -0.25,3 -0.1767766953,2.8232233047 0,2.75 0.1767766953,2.8232233047\\"></polygon><polygon points=\\"0.25,3.6 0.1915111108,3.7606969024 0.0434120444,3.8462019383 -0.125,3.8165063509 -0.2349231552,3.6855050358 -0.2349231552,3.5144949642 -0.125,3.3834936491 0.0434120444,3.3537980617 0.1915111108,3.4393030976\\"></polygon><polygon points=\\"0.25,4.2 0.2022542486,4.3469463131 0.0772542486,4.4377641291 -0.0772542486,4.4377641291 -0.2022542486,4.3469463131 -0.25,4.2 -0.2022542486,4.0530536869 -0.0772542486,3.9622358709 0.0772542486,3.9622358709 0.2022542486,4.0530536869\\"></polygon><polygon points=\\"0.25,4.8 0.2103133832,4.9351602044 0.1038537533,5.0274079988 -0.0355787096,5.0474553605 -0.1637151835,4.9889373936 -0.2398732434,4.8704331392 -0.2398732434,4.7295668608 -0.1637151835,4.6110626064 -0.0355787096,4.5525446395 0.1038537533,4.5725920012 0.2103133832,4.6648397956\\"></polygon><polygon points=\\"0.25,5.4 0.2165063509,5.525 0.125,5.6165063509 0,5.65 -0.125,5.6165063509 -0.2165063509,5.525 -0.25,5.4 -0.2165063509,5.275 -0.125,5.1834936491 0,5.15 0.125,5.1834936491 0.2165063509,5.275\\"></polygon></g></svg>"`;

exports[`@tracespace/cli render > should render pads/rectangle.gbr > parse 1`] = `
{
  "children": [
    {
      "format": [
        3,
        4,
      ],
      "mode": "absolute",
      "position": {
        "end": {
          "column": 12,
          "line": 1,
          "offset": 11,
        },
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "coordinateFormat",
      "zeroSuppression": "leading",
    },
    {
      "position": {
        "end": {
          "column": 6,
          "line": 2,
          "offset": 19,
        },
        "start": {
          "column": 2,
          "line": 2,
          "offset": 15,
        },
      },
      "type": "units",
      "units": "in",
    },
    {
      "code": "10",
      "hole": null,
      "position": {
        "end": {
          "column": 16,
          "line": 3,
          "offset": 37,
        },
        "start": {
          "column": 2,
          "line": 3,
          "offset": 23,
        },
      },
      "shape": {
        "type": "rectangle",
        "xSize": 1,
        "ySize": 0.5,
      },
      "type": "toolDefinition",
    },
    {
      "code": "10",
      "position": {
        "end": {
          "column": 4,
          "line": 4,
          "offset": 43,
        },
        "start": {
          "column": 1,
          "line": 4,
          "offset": 40,
        },
      },
      "type": "toolChange",
    },
    {
      "coordinates": {
        "x": "0",
        "y": "0",
      },
      "graphic": "shape",
      "position": {
        "end": {
          "column": 8,
          "line": 5,
          "offset": 52,
        },
        "start": {
          "column": 1,
          "line": 5,
          "offset": 45,
        },
      },
      "type": "graphic",
    },
    {
      "position": {
        "end": {
          "column": 4,
          "line": 6,
          "offset": 57,
        },
        "start": {
          "column": 1,
          "line": 6,
          "offset": 54,
        },
      },
      "type": "done",
    },
  ],
  "filetype": "gerber",
  "type": "root",
}
`;

exports[`@tracespace/cli render > should render pads/rectangle.gbr > plot 1`] = `
{
  "children": [
    {
      "children": [
        {
          "shape": {
            "r": null,
            "type": "rectangle",
            "x": -0.5,
            "xSize": 1,
            "y": -0.25,
            "ySize": 0.5,
          },
          "type": "imageShape",
        },
      ],
      "size": [
        -0.5,
        -0.25,
        0.5,
        0.25,
      ],
      "type": "imageLayer",
    },
  ],
  "type": "image",
  "units": "in",
}
`;

exports[`@tracespace/cli render > should render pads/rectangle.gbr > render 1`] = `"<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"0\\" fill-rule=\\"evenodd\\" fill=\\"currentColor\\" stroke=\\"currentColor\\" width=\\"1in\\" height=\\"0.5in\\" viewBox=\\"-0.5 -0.25 1 0.5\\"><g transform=\\"translate(0, 0) scale(1,-1)\\"><rect x=\\"-0.5\\" y=\\"-0.25\\" width=\\"1\\" height=\\"0.5\\"></rect></g></svg>"`;
